Week3/MAKEME.md

```diff
@@ -7,170 +7,120 @@ Topics discussed this week:
-  const hyfReposUrl = hyfUrl + '/orgs/HackYourFuture/repos?per_page=100';
-
-  class Repository {
-    constructor(data) {
-      this._data = data;
-    }
-
-    /**
-     * Render the repository info to the DOM.
-     * @param {HTML element} parent The parent element in which to render the repository.
-     * info.
-     */
-    render(parent) {
-      // Add your code here.
-      // This method should render the repository data stored in the '_data' property
-      // as HTML elements and append them to the `parent` element.
-    }
-
-    /**
-     * Returns an array of contributors as a promise
-    */
-    fetchContributors() {
-      // Add your code here
-    }
-  }
-
-  class Contributor {
-    constructor(data) {
-      this._data = data;
-    }
-
-    /**
-     * Render the contributor info to the DOM.
-     * @param {HTML element} parent The parent element in which to render the contributor.
-     * info.
-     */
-    render(parent) {
-      // Add your code here.
-      // This method should render the contributor data stored in the 'data' property
-      // as HTML elements and append them to the `parent` element.
-    }
-  }
-
-  class View {
-    constructor() {
-      this.initialize();
-    }
-
-    /**
-     * View initialization
-     */
-    async initialize() {
-      // Add code here to initialize your app
-      // 1. Create the fixed HTML elements of your page
-      // 2. Make an initial XMLHttpRequest to populate your <select> element
-    }
-
-    /**
-     * Fetch information for the selected repository and render the
-     * information as HTML elements in the DOM
-     * @param {*} repoName The name of the selected repository
-     */
-    async fetchAndRender(repoName) {
-      const leftDiv = ...;
-      const rightDiv = ...;
-
-      // ...
-
-      try {
-        const contributors = await repo.fetchContributors();
-        repo.render(leftDiv);
-        contributors
-          .map(contributor => new Contributor(contributor))
-          .forEach(contributor => contributor.render(contributorList));
-      }
-      catch (error) {
-        createAndAppend('div', container, { html: error.message, class: 'alert alert-error' });
-      }
-    }
-
-    /**
-     * Fetch API data as JSON data in a promise
-     * @param {string} url The URL of the API endpoint.
-     * @returns A promise.
-     */
-    fetchJSON(url) {
-      // Add your code here
-    }
-  }
-
-  window.onload = () => new View();
-}
-```
+1. Refactor all `.then()` and `.catch()` methods with `async`/`await` and `try...catch`.
+2. Make your app ARIA-compliant (see below).
+3. Refactor your application to use ES6 classes.
+
+
+#### 2.2.1 async/await
+
+**Instructions:**
+
+1. Refactor all `.then()` and `.catch()` methods with `async`/`await` and `try...catch`.
+
+2. Make sure that your error handling code still works. See the week2 MAKEME on how to force an error response from GitHub.
+
+
+#### 2.2.2 ARIA-compliance
```

Comment:
could you please turn 2.2.2 and 2.2.3 around? And make 2.2.2 the bonus assignment? I'm afraid that this is a lot of homework for one week and that students will not be able to finish all and will leave out 2.2.3

Reply:
2.2.2 is now ES6 Classes (BONUS)
2.2.3 is now ARIA-compliance